sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -139,3 +140,17 @@ def get_next_group_id(self):
 
     self.group_id += 1
     return self.group_id
+
+  # The following functions are used by the users to customize pipeline level configurations
+  # Note: call these function inside the pipeline python function.
+  @staticmethod
+  def set_image_pull_secrets(image_pull_secrets):
```

Comment:
Let's not expose the private `Pipeline` class to the users.

Bradley did not want it to be used by pipeline authors:
https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_pipeline.py#L43

Reply:
fixed in the new commit.
Pipeline instance is hidden.